pubsub/transport/amqp/connection.go

```diff
@@ -109,11 +110,22 @@ func (ch *Channel) Consume(queue, consumer string, autoAck, exclusive, noLocal,
 			}
 
 			// sleep before IsClose call. closed flag may not set before sleep.
-			time.Sleep(delay * time.Second)
 
 			if ch.IsClosed() {
 				break
 			}
+
+			done := false
+			select {
+			case <-ctx.Done():
```

Comment:
this will work only when you stopped the transport, but you have removed the transport stopping call from the subscriber, so if you canceled the context of the subscriber without transport being stopped this goroutine will continue hanging, the loop above of deliveries channel will keep this goroutine blocked and alive.

Reply:
outdated since we decided to stop only by closing the context